modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductConfigurations.testcase

```diff
@@ -1,28 +1,166 @@
 @component-name = "portal-commerce"
 definition {
 
+	property custom.properties = "jsonws.web.service.paths.excludes=";
 	property portal.release = "true";
 	property portal.upstream = "true";
 	property release.feature.flags.disable.DISABLE_PRIVATE_LAYOUTS = "true";
 	property testray.main.component.name = "Product Info Management";
 
 	setUp {
-		CommerceConfiguration.commerceSetUp(storeFront = "true");
+		CommerceConfiguration.commerceSetUp(
+			minium = "true",
+			storeFront = "true");
 	}
 
 	tearDown {
 		CommerceConfiguration.commerceTearDown();
 	}
 
+	@description = "This is a test for COMMERCE-10491. The product can be added to cart until there's stock quantity"
+	@priority = "3"
+	test CanOnlyPlaceOrderWhenStockIsAvailable {
+		property portal.acceptance = "true";
+
+		task ("Given a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And given the warehouse quanitity for the products is set to 4") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceEntry.editWarehousesQuantity(
+				italyQuantity = "1",
+				northeastQuantity = "1",
+				productName = "U-Joint",
+				productSku = "MIN55861",
+				southwestQuantity = "2");
+		}
+
+		task ("And product's Allow Back Order is disabled") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				allowBackOrder = "false",
+				productName = "U-Joint");
+		}
+
+		task ("And buyer user logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("And given a placed order") {
```

Comment:
You could use json macro to place an order, if you don't have any specific reason to do it from the sf.

Reply:
We do it in the SF because we want to see the interactions with the "Add to the cart" button and how he is going to behave.